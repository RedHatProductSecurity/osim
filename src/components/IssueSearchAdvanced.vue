<script setup lang="ts">
import { computed, ref } from 'vue';
import { flawImpacts, flawSources, flawIncidentStates } from '@/types/zodFlaw';
import { useRoute } from 'vue-router';
import { flawFields } from '@/constants/flawFields';
import { useSearchParams } from '@/composables/useSearchParams';
import { descriptionRequiredStates } from '@/types/zodFlaw';
import { affectAffectedness } from '@/types/zodAffect';
import { sort } from 'ramda';
const { facets, removeFacet, submitAdvancedSearch } = useSearchParams();

const props = defineProps<{
  isLoading: boolean;
}>();

const route = useRoute();

const emit = defineEmits(['filter:save']);

const nameForOption = (fieldName: string) => {
  const mappings: Record<string, string> = {
    uuid: 'UUID',
    cvss_scores__score: 'CVSS Score',
    cvss_scores__vector: 'CVSS Vector',
    affects__ps_module: 'Affected Module',
    affects__ps_component: 'Affected Component',
    affects__affectedness: 'Affected Affectedness',
    affects__trackers__ps_update_stream: 'Affect Update Stream',
    acknowledgments__name: 'Acknowledgment Author',
    affects__trackers__errata__advisory_name: 'Errata Advisory Name',
    affects__trackers__external_system_id: 'Tracker External System ID',
    workflow_state: 'Flaw State',
    cwe_id: 'CWE ID',
    cve_id: 'CVE ID',
    source: 'CVE Source',
    cve_description: 'CVE Description',
    requires_cve_description: 'CVE Description Review',
    major_incident_state: 'Incident State',
  };
  let name =
    mappings[fieldName]
    || fieldName.replace(/__[a-z]/g, (label) => `: ${label.charAt(2).toUpperCase()}`);
  name = name.replace(/_/g, ' ');
  return name.charAt(0).toUpperCase() + name.slice(1);
};

const sortFieldNames = sort((fieldA: string, fieldB: string) =>
  nameForOption(fieldA).localeCompare(nameForOption(fieldB))
);

const chosenFields = computed(() => facets.value.map(({ field }) => field));

const unchosenFields = (chosenField: string) =>
  sortFieldNames(flawFields.filter((field) => !chosenFields.value.includes(field) || field === chosenField));


const optionsFor = (field: string) =>
  ({
    source: flawSources,
    impact: flawImpacts,
    requires_cve_description: descriptionRequiredStates,
    embargoed: ['true', 'false'],
    workflow_state: [
      'DONE',
      'NEW',
      'PRE_SECONDARY_ASSESSMENT',
      'REJECTED',
      'SECONDARY_ASSESSMENT',
      'TRIAGE',
    ],
    major_incident_state: flawIncidentStates,
    affects__affectedness: affectAffectedness,
  })[field] || null;
const shouldShowAdvanced = ref(route.query.mode === 'advanced');
</script>

<template>
  <details :open="shouldShowAdvanced" class="osim-advanced-search-container">
    <summary class="mb-2" @click="shouldShowAdvanced = true">Advanced Search</summary>
    <form class="mb-2" @submit.prevent="submitAdvancedSearch">
      <div v-for="(facet, index) in facets" :key="facet.field" class="input-group mb-2">
        <select v-model="facet.field" class="form-select search-facet-field" @submit.prevent>
          <option
            v-if="!facet.field"
            selected
            value=""
            disabled
          >Select field...</option>
          <option v-for="field in unchosenFields(facet.field)" :key="field" :value="field">
            {{ nameForOption(field) }}
          </option>
        </select>
        <select
          v-if="optionsFor(facet.field)"
          v-model="facet.value"
          class="form-select"
          @submit.prevent
        >
          <option v-for="option in optionsFor(facet.field)" :key="option" :value="option">
            {{ option }}
          </option>
        </select>
        <input
          v-else
          v-model="facet.value"
          type="text"
          class="form-control"
          :disabled="!facet.field"
        />
        <button class="btn btn-outline-primary" type="button" @click="removeFacet(index)">
          <i class="bi-x" aria-label="remove field"></i>
        </button>
      </div>
      <button class="btn btn-primary me-3" type="submit" :disabled="props.isLoading">
        <div v-if="props.isLoading" class="spinner-border spinner-border-sm"></div>
        Search
      </button>
      <button
        class="btn btn-primary me-3"
        type="button"
        :disabled="isLoading"
        @click="emit('filter:save')"
      >
        Save as Default
      </button>
    </form>
  </details>
</template>

<style lang="scss" scoped>
.osim-advanced-search-container {
  i {
    cursor: pointer;
  }

  select.search-facet-field {
    display: flex;
    width: auto;
    flex-grow: 0;
  }
}
</style>
