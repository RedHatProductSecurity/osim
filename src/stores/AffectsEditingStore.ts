import { ref, computed, onUnmounted, type Ref } from 'vue';

import { clone } from 'ramda';
import { defineStore } from 'pinia';

import { useFlaw } from '@/composables/useFlaw';
import { useFlawAffectsModel } from '@/composables/useFlawAffectsModel';

import type { ZodAffectType } from '@/types/zodAffect';
import { matcherForAffect } from '@/utils/helpers';

export const useAffectsEditingStore = defineStore('EditableAffects', () => {
  const { flaw } = useFlaw();

  const { initialAffects } = useFlawAffectsModel();
  const affectSelections = ref<ZodAffectType[]>([]);

  const {
    areAllAffectsSelectable,
    areAllAffectsSelected,
    isAffectSelected,
    isIndeterminateSelection,
    isSelectable,
    resetSelections,
    selectedAffects,
    toggleAffectSelection,
    toggleMultipleAffectSelections,
  } = useAffectSelections(affectSelections, isBeingEdited);

  const affectValuesPriorEdit = ref<ZodAffectType[]>([]);
  const affectsBeingEdited = ref<ZodAffectType[]>([]);

  onUnmounted($reset);

  function isBeingEdited(affect: ZodAffectType) {
    const matchingAffect = affectsBeingEdited.value.find(matcherForAffect(affect)) as ZodAffectType;
    return affectsBeingEdited.value.includes(matchingAffect);
  }

  function getAffectPriorEdit(affect: ZodAffectType): ZodAffectType {
    return affectValuesPriorEdit.value.find(prior => prior.uuid && prior.uuid === affect.uuid) || affect;
  }

  function editAffect(affect: ZodAffectType) {
    if (isAffectSelected(affect)) {
      toggleAffectSelection(affect);
    }

    if (!isBeingEdited(affect)) {
      const affectToEdit = flaw.value.affects.find(matcherForAffect(affect));
      if (affectToEdit) {
        affectsBeingEdited.value.push(affectToEdit);
        const affectCopy = clone(affect) as ZodAffectType;
        affectValuesPriorEdit.value.push(affectCopy);
      }
    }
  }

  function commitChanges(affect: ZodAffectType) {
    const matchAffect = matcherForAffect(affect);
    const updateIndex = flaw.value.affects.findIndex(matchAffect);
    flaw.value.affects.splice(updateIndex, 1, affect);
    resetStagedAffectEdit(affect);
  }

  function cancelChanges(affect: ZodAffectType) {
    resetStagedAffectEdit(affect);
  }

  function revertAffectToLastSaved(affect: ZodAffectType) {
    const saved = initialAffects.value.find(a => a.uuid === affect.uuid);
    const index = flaw.value.affects.findIndex(a => a.uuid === affect.uuid);
    if (index !== -1 && saved) {
      flaw.value.affects[index] = { ...saved };
    }
    if (isAffectSelected(affect)) {
      toggleAffectSelection(affect);
    }
  }

  function resetStagedAffectEdit(affect: ZodAffectType) {
    const matchAffect = matcherForAffect(affect);
    const editingIndex = affectsBeingEdited.value.findIndex(matchAffect);
    affectsBeingEdited.value.splice(editingIndex, 1);
    affectValuesPriorEdit.value.splice(affectValuesPriorEdit.value.findIndex(matchAffect), 1);
    if (isAffectSelected(affect)) {
      toggleAffectSelection(affect);
    }
  }
  function editSelectedAffects() {
    selectedAffects.value.forEach(editAffect);
  }

  function commitAllChanges() {
    const affectsToCommit = [...affectsBeingEdited.value];
    affectsToCommit.forEach(commitChanges);
  }

  function cancelAllChanges() {
    const affectsToCancel = [...affectsBeingEdited.value];
    affectsToCancel.forEach(cancelChanges);
  }

  function $reset() {
    affectsBeingEdited.value = [];
    affectValuesPriorEdit.value = [];
  }

  return {
    cancelChanges,
    commitChanges,
    isBeingEdited,
    getAffectPriorEdit,
    editAffect,
    revertAffectToLastSaved,
    affectValuesPriorEdit,
    affectsBeingEdited,
    $reset,
    selectedAffects,
    areAllAffectsSelected,
    isIndeterminateSelection,
    isSelectable,
    areAllAffectsSelectable,
    isAffectSelected,
    toggleAffectSelection,
    toggleMultipleAffectSelections,
    resetSelections,
    editSelectedAffects,
    commitAllChanges,
    cancelAllChanges,
  };
});

function useAffectSelections(selectedAffects: Ref<ZodAffectType[]>, isBeingEdited: (affect: ZodAffectType) => boolean) {
  const { flaw } = useFlaw();
  const affects = computed(() => flaw.value.affects);

  const { isAffectBeingRemoved } = useFlawAffectsModel();

  function isSelectable(affect: ZodAffectType) {
    return !isAffectBeingRemoved(affect) && !isBeingEdited(affect);
  }

  function isAffectSelected(affect: ZodAffectType) {
    return isSelectable(affect) && selectedAffects.value.includes(affect);
  }

  const areAllAffectsSelected = computed(() => {
    return affects.value.every(isAffectSelected);
  });

  const isIndeterminateSelection = computed(() => {
    return !areAllAffectsSelected.value && affects.value.some(isAffectSelected);
  });

  const areAllAffectsSelectable = computed(() => affects.value.every(isSelectable));

  function toggleAffectSelection(affect: ZodAffectType) {
    if (!isSelectable(affect)) {
      return;
    }
    if (!isAffectSelected(affect)) {
      selectedAffects.value.push(affect);
    } else {
      selectedAffects.value = selectedAffects.value.filter(a => a !== affect);
    }
  }

  function toggleMultipleAffectSelections(event: Event) {
    if (areAllAffectsSelected.value) {
      affects.value.forEach(affect => toggleAffectSelection(affect));
    } else if (selectedAffects.value.length === 0) {
      affects.value.filter(affect => !isAffectSelected(affect)).forEach(affect => toggleAffectSelection(affect));
    } else {
      // De-select all selected affects if only some are selected
      affects.value.filter(affect => isAffectSelected(affect)).forEach(toggleAffectSelection);
      (event.target as HTMLInputElement).checked = false;
    }
  }

  function resetSelections() {
    selectedAffects.value = [];
  }

  return {
    selectedAffects,
    areAllAffectsSelected,
    isIndeterminateSelection,
    isSelectable,
    areAllAffectsSelectable,
    isAffectSelected,
    toggleAffectSelection,
    toggleMultipleAffectSelections,
    resetSelections,
  };
}
