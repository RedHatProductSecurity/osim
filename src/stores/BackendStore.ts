import { ref } from 'vue';
import { defineStore } from 'pinia';


export interface OsimBackends {
  osidb: string | unknown;
  isReady: () => boolean;
  error: boolean | string;
}

export const useBackendStore = defineStore('BackendStore', () => {
  const error = ref<string>('');
  const osidb = ref<string>('');

  function isReady() {
    return osidb.value !== '';
  }

  return {
    error,
    osidb,
    isReady,
  };
});
