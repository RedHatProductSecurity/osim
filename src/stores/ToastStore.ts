import { ref } from 'vue';
import { defineStore } from 'pinia';
import { DateTime } from 'luxon';


export interface ToastNew {
  title?: string,
  body: string,
  // timestamp: moment.Moment,
  // key?: number,
  bodyHtml?: boolean,
  timeoutMs?: number,
  css?: 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark',
}

interface ToastAdded extends ToastNew {
  id: number,
  timestamp: DateTime,
  isFresh: boolean,
}

let toastId = 0;

export const useToastStore = defineStore('ToastStore', () => {
  const toasts = ref<ToastAdded[]>([]);

  function addToast(toast: ToastNew) {
    const newToast: ToastAdded = {
      ...toast,
      id: toastId++,
      timestamp: DateTime.now(),
      isFresh: true,
    };
    toasts.value.unshift(newToast);
  }

  function $reset() {
    toasts.value = [];
  }

  return {
    $reset,
    addToast,
    toasts,
  };
});

