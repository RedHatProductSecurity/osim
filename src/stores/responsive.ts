// Track header and footer for elements between to have responsive positioning

import { ref } from 'vue';

const navbarHeight = ref<number>(0);
const navbarBottom = ref<number>(0);

const footerTop = ref<number>(0);
const footerHeight = ref<number>(0);

export {
  navbarHeight,
  navbarBottom,
  footerTop,
  footerHeight,
};
