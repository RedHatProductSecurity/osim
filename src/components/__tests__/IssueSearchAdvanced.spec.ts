import { mount, VueWrapper } from '@vue/test-utils';
import { describe, it, expect, vi, type Mock } from 'vitest';
import { useRoute } from 'vue-router';
import IssueSearchAdvanced from '@/components/IssueSearchAdvanced.vue';
import { useToastStore } from '@/stores/ToastStore';

import { createTestingPinia } from '@pinia/testing';


vi.mock('vue-router', async () => {
  const actual = await vi.importActual('vue-router');
  return {
    // eslint-disable-next-line @typescript-eslint/ban-types
    ...actual as {},
    useRoute: vi.fn(),
  };
});

(useRoute as Mock).mockReturnValue({
  'query': { mode: 'advanced' },
});

describe('IssueSearchAdvanced', () => {
  let wrapper: VueWrapper<any>;

  beforeAll(() => {
    createTestingPinia({
      initialState: {
        toasts: [],
      },
    });
  });

  beforeEach(() => {
    const props: typeof IssueSearchAdvanced.props = {};
    wrapper = mount(IssueSearchAdvanced, {
      props,
      plugins:[useToastStore()],
      global: {
        mocks: { useRoute },
      }
    });
  });

  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
