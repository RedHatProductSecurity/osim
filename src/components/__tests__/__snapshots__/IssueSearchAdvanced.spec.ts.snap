// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`issueSearchAdvanced > should add new facet when last facet is populated 1`] = `
"<details data-v-b86992a8="" open="" class="osim-advanced-search-container">
  <summary data-v-b86992a8="" class="mb-1">Advanced Search</summary>
  <form data-v-b86992a8="" class="mb-2">
    <div data-v-b86992a8="" class="input-group my-1">
      <div data-v-b86992a8="" class="query-input form-control bg-secondary text-white"><span data-v-b86992a8="" class="my-auto">Query Filter</span><button data-v-b86992a8="" class="btn btn-sm text-white" type="button"><i data-v-b86992a8="" class="bi bi-question-circle-fill fs-5" aria-label="hide query filter"></i></button></div><textarea data-v-b86992a8="" class="form-control" style="height: 38px;"></textarea><button data-v-b86992a8="" disabled="" class="btn btn-sm btn-secondary rounded-0 py-0" title="Clear query" type="button"><i data-v-b86992a8="" class="bi bi-eraser fs-5"></i></button><button data-v-b86992a8="" class="btn btn-secondary py-0" type="button"><i data-v-b86992a8="" class="bi bi-x fs-5" aria-label="hide query filter"></i></button>
    </div>
    <div data-v-b86992a8="" class="input-group my-1"><select data-v-b86992a8="" class="form-select search-facet-field">
        <!--v-if-->
        <option data-v-b86992a8="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-b86992a8="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-b86992a8="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-b86992a8="" value="affects__ps_component">Affected Component</option>
        <option data-v-b86992a8="" value="affects__ps_module">Affected Module</option>
        <option data-v-b86992a8="" value="cve_description">CVE Description</option>
        <option data-v-b86992a8="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-b86992a8="" value="cve_id">CVE ID</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="embargoed">Embargoed</option>
        <option data-v-b86992a8="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-b86992a8="" value="workflow_state">Flaw State</option>
        <option data-v-b86992a8="" value="impact">Impact</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="mitigation">Mitigation</option>
        <option data-v-b86992a8="" value="owner">Owner</option>
        <option data-v-b86992a8="" value="statement">Statement</option>
        <option data-v-b86992a8="" value="title">Title</option>
        <option data-v-b86992a8="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-b86992a8="" value="uuid">UUID</option>
      </select><input data-v-b86992a8="" type="text" class="form-control">
      <div data-v-b86992a8="" class="btn-group"><button data-v-b86992a8="" class="btn btn-sm btn-primary rounded-0 py-0" title="Empty field search" type="button"><i data-v-b86992a8="" class="bi bi-circle fs-5"></i></button><button data-v-b86992a8="" class="btn btn-sm btn-primary rounded-0 py-0" title="Non empty field search" type="button"><i data-v-b86992a8="" class="bi bi-slash-circle fs-5"></i></button></div><button data-v-b86992a8="" class="btn btn-sm btn-primary rounded-0 py-0" title="Clear field" type="button"><i data-v-b86992a8="" class="bi bi-eraser fs-5"></i></button><button data-v-b86992a8="" class="btn btn-primary py-0" type="button"><i data-v-b86992a8="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-b86992a8="" class="input-group my-1"><select data-v-b86992a8="" class="form-select search-facet-field">
        <option data-v-b86992a8="" selected="" disabled="" value="">Select field...</option>
        <option data-v-b86992a8="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-b86992a8="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-b86992a8="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-b86992a8="" value="affects__ps_component">Affected Component</option>
        <option data-v-b86992a8="" value="affects__ps_module">Affected Module</option>
        <option data-v-b86992a8="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-b86992a8="" value="cve_id">CVE ID</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="embargoed">Embargoed</option>
        <option data-v-b86992a8="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-b86992a8="" value="workflow_state">Flaw State</option>
        <option data-v-b86992a8="" value="impact">Impact</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="mitigation">Mitigation</option>
        <option data-v-b86992a8="" value="owner">Owner</option>
        <option data-v-b86992a8="" value="statement">Statement</option>
        <option data-v-b86992a8="" value="title">Title</option>
        <option data-v-b86992a8="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-b86992a8="" value="uuid">UUID</option>
      </select><input data-v-b86992a8="" type="text" class="form-control" disabled="">
      <!--v-if-->
      <!--v-if--><button data-v-b86992a8="" class="btn btn-primary py-0" type="button"><i data-v-b86992a8="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-b86992a8="" class="d-flex mt-2"><button data-v-b86992a8="" class="btn btn-primary me-2" type="submit" style="width: 23ch;" aria-label="Advance search button">
        <!--v-if--> Search
      </button>
      <!--v-if--><select data-v-b86992a8="" class="sort-by-select form-select search-facet-field" title="When using this sorting with active table column sort, the table column field will be used as secondary sorting key">
        <option data-v-b86992a8="" selected="" disabled="" label="Sort By..."></option>
        <option data-v-b86992a8="" hidden="" label="" value=""></option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
      </select>
      <!--v-if-->
    </div>
  </form>
</details>
<details data-v-b86992a8="" open="" class="osim-advanced-search-container">
  <summary data-v-b86992a8=""> Saved Searches</summary>
  <div data-v-b86992a8="" class="container-fluid mt-2"><button data-v-b86992a8="" title="Query: django query
Fields: affects__ps_component: test" class="btn me-2 mt-1 border" type="button">name <i data-v-b86992a8="" class="bi ms-2 bi-star"></i></button></div>
  <!--v-if-->
  <transition-stub data-v-296c460d="" name="modal" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
  <transition-stub data-v-296c460d="" name="modal-bg" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
  <div data-v-b86992a8="" class="d-flex mt-2"><button data-v-b86992a8="" class="btn btn-primary me-2" aria-label="Save search" type="button"> Save Search </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</details>
<transition-stub data-v-296c460d="" name="modal" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>
<transition-stub data-v-296c460d="" name="modal-bg" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>"
`;

exports[`issueSearchAdvanced > should render when \`isLoading\` is false 1`] = `
"<details data-v-b86992a8="" open="" class="osim-advanced-search-container">
  <summary data-v-b86992a8="" class="mb-1">Advanced Search</summary>
  <form data-v-b86992a8="" class="mb-2">
    <div data-v-b86992a8="" class="input-group my-1">
      <div data-v-b86992a8="" class="query-input form-control bg-secondary text-white"><span data-v-b86992a8="" class="my-auto">Query Filter</span><button data-v-b86992a8="" class="btn btn-sm text-white" type="button"><i data-v-b86992a8="" class="bi bi-question-circle-fill fs-5" aria-label="hide query filter"></i></button></div><textarea data-v-b86992a8="" class="form-control" style="height: 38px;"></textarea><button data-v-b86992a8="" disabled="" class="btn btn-sm btn-secondary rounded-0 py-0" title="Clear query" type="button"><i data-v-b86992a8="" class="bi bi-eraser fs-5"></i></button><button data-v-b86992a8="" class="btn btn-secondary py-0" type="button"><i data-v-b86992a8="" class="bi bi-x fs-5" aria-label="hide query filter"></i></button>
    </div>
    <div data-v-b86992a8="" class="input-group my-1"><select data-v-b86992a8="" class="form-select search-facet-field">
        <option data-v-b86992a8="" selected="" disabled="" value="">Select field...</option>
        <option data-v-b86992a8="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-b86992a8="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-b86992a8="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-b86992a8="" value="affects__ps_component">Affected Component</option>
        <option data-v-b86992a8="" value="affects__ps_module">Affected Module</option>
        <option data-v-b86992a8="" value="cve_description">CVE Description</option>
        <option data-v-b86992a8="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-b86992a8="" value="cve_id">CVE ID</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="embargoed">Embargoed</option>
        <option data-v-b86992a8="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-b86992a8="" value="workflow_state">Flaw State</option>
        <option data-v-b86992a8="" value="impact">Impact</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="mitigation">Mitigation</option>
        <option data-v-b86992a8="" value="owner">Owner</option>
        <option data-v-b86992a8="" value="statement">Statement</option>
        <option data-v-b86992a8="" value="title">Title</option>
        <option data-v-b86992a8="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-b86992a8="" value="uuid">UUID</option>
      </select><input data-v-b86992a8="" type="text" class="form-control" disabled="">
      <!--v-if-->
      <!--v-if--><button data-v-b86992a8="" class="btn btn-primary py-0" type="button"><i data-v-b86992a8="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-b86992a8="" class="d-flex mt-2"><button data-v-b86992a8="" class="btn btn-primary me-2" type="submit" style="width: 23ch;" aria-label="Advance search button">
        <!--v-if--> Search
      </button>
      <!--v-if--><select data-v-b86992a8="" class="sort-by-select form-select search-facet-field" title="When using this sorting with active table column sort, the table column field will be used as secondary sorting key">
        <option data-v-b86992a8="" selected="" disabled="" label="Sort By..."></option>
        <option data-v-b86992a8="" hidden="" label="" value=""></option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
      </select>
      <!--v-if-->
    </div>
  </form>
</details>
<details data-v-b86992a8="" open="" class="osim-advanced-search-container">
  <summary data-v-b86992a8=""> Saved Searches</summary>
  <div data-v-b86992a8="" class="container-fluid mt-2"><button data-v-b86992a8="" title="Query: django query
Fields: affects__ps_component: test" class="btn me-2 mt-1 border" type="button">name <i data-v-b86992a8="" class="bi ms-2 bi-star"></i></button></div>
  <!--v-if-->
  <transition-stub data-v-296c460d="" name="modal" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
  <transition-stub data-v-296c460d="" name="modal-bg" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
  <div data-v-b86992a8="" class="d-flex mt-2"><button data-v-b86992a8="" class="btn btn-primary me-2" aria-label="Save search" type="button" disabled=""> Save Search </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</details>
<transition-stub data-v-296c460d="" name="modal" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>
<transition-stub data-v-296c460d="" name="modal-bg" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>"
`;

exports[`issueSearchAdvanced > should render when \`isLoading\` is true 1`] = `
"<details data-v-b86992a8="" open="" class="osim-advanced-search-container">
  <summary data-v-b86992a8="" class="mb-1">Advanced Search</summary>
  <form data-v-b86992a8="" class="mb-2">
    <div data-v-b86992a8="" class="input-group my-1">
      <div data-v-b86992a8="" class="query-input form-control bg-secondary text-white"><span data-v-b86992a8="" class="my-auto">Query Filter</span><button data-v-b86992a8="" class="btn btn-sm text-white" type="button"><i data-v-b86992a8="" class="bi bi-question-circle-fill fs-5" aria-label="hide query filter"></i></button></div><textarea data-v-b86992a8="" class="form-control" style="height: 38px;"></textarea><button data-v-b86992a8="" disabled="" class="btn btn-sm btn-secondary rounded-0 py-0" title="Clear query" type="button"><i data-v-b86992a8="" class="bi bi-eraser fs-5"></i></button><button data-v-b86992a8="" class="btn btn-secondary py-0" type="button"><i data-v-b86992a8="" class="bi bi-x fs-5" aria-label="hide query filter"></i></button>
    </div>
    <div data-v-b86992a8="" class="input-group my-1"><select data-v-b86992a8="" class="form-select search-facet-field">
        <option data-v-b86992a8="" selected="" disabled="" value="">Select field...</option>
        <option data-v-b86992a8="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-b86992a8="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-b86992a8="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-b86992a8="" value="affects__ps_component">Affected Component</option>
        <option data-v-b86992a8="" value="affects__ps_module">Affected Module</option>
        <option data-v-b86992a8="" value="cve_description">CVE Description</option>
        <option data-v-b86992a8="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-b86992a8="" value="cve_id">CVE ID</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="embargoed">Embargoed</option>
        <option data-v-b86992a8="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-b86992a8="" value="workflow_state">Flaw State</option>
        <option data-v-b86992a8="" value="impact">Impact</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="mitigation">Mitigation</option>
        <option data-v-b86992a8="" value="owner">Owner</option>
        <option data-v-b86992a8="" value="statement">Statement</option>
        <option data-v-b86992a8="" value="title">Title</option>
        <option data-v-b86992a8="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-b86992a8="" value="uuid">UUID</option>
      </select><input data-v-b86992a8="" type="text" class="form-control" disabled="">
      <!--v-if-->
      <!--v-if--><button data-v-b86992a8="" class="btn btn-primary py-0" type="button"><i data-v-b86992a8="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-b86992a8="" class="d-flex mt-2"><button data-v-b86992a8="" class="btn btn-primary me-2" type="submit" style="width: 23ch;" disabled="" aria-label="Advance search button">
        <div data-v-b86992a8="" class="spinner-border spinner-border-sm"></div> Search
      </button>
      <!--v-if--><select data-v-b86992a8="" class="sort-by-select form-select search-facet-field" title="When using this sorting with active table column sort, the table column field will be used as secondary sorting key">
        <option data-v-b86992a8="" selected="" disabled="" label="Sort By..."></option>
        <option data-v-b86992a8="" hidden="" label="" value=""></option>
        <option data-v-b86992a8="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-b86992a8="" value="cwe_id">CWE ID</option>
        <option data-v-b86992a8="" value="major_incident_state">Incident State</option>
        <option data-v-b86992a8="" value="source">CVE Source</option>
      </select>
      <!--v-if-->
    </div>
  </form>
</details>
<details data-v-b86992a8="" open="" class="osim-advanced-search-container">
  <summary data-v-b86992a8=""> Saved Searches</summary>
  <div data-v-b86992a8="" class="container-fluid mt-2"><button data-v-b86992a8="" title="Query: django query
Fields: affects__ps_component: test" class="btn me-2 mt-1 border" type="button">name <i data-v-b86992a8="" class="bi ms-2 bi-star"></i></button></div>
  <!--v-if-->
  <transition-stub data-v-296c460d="" name="modal" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
  <transition-stub data-v-296c460d="" name="modal-bg" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
  <div data-v-b86992a8="" class="d-flex mt-2"><button data-v-b86992a8="" class="btn btn-primary me-2" aria-label="Save search" type="button" disabled=""> Save Search </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</details>
<transition-stub data-v-296c460d="" name="modal" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>
<transition-stub data-v-296c460d="" name="modal-bg" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>"
`;
